app/src/main/java/com/duckduckgo/app/browser/BrowserTabViewModel.kt

```diff
@@ -4179,6 +4179,23 @@ class BrowserTabViewModel @Inject constructor(
         lastSubmittedUserQuery = query
     }
 
+    fun onExperimentalHomeScreenWidgetBottomSheetDialogShown() {
+        logcat { "Experimental Home Screen Widget bottom sheet shown" }
+    }
+
+    fun onExperimentalHomeScreenWidgetBottomSheetDialogCancelled() {
+        logcat { "Experimental Home Screen Widget bottom sheet dismissed" }
+    }
+
+    fun onExperimentalHomeScreenWidgetBottomSheetDialogAddWidgetClicked() {
+        logcat { "Add Widget clicked on the Experimental Home Screen Widget bottom sheet" }
+        command.value = LaunchAddWidget
+    }
+
+    fun onExperimentalHomeScreenWidgetBottomSheetDialogNotNowClicked() {
+        logcat { "Not Now clicked on the Experimental Home Screen Widget bottom sheet" }
+    }
```

Comment:
When testing, the widget dialog keeps reappearing every time I open a new tab after dismissing it. You need to ensure that you call ctaViewModel.onUserDismissedCta() to make sure it's recorded in the dismissed CTA database.
Same for other callbacks that dismiss the CTA.